pkg/utils/get_ipfamily.go

```diff
@@ -26,3 +33,88 @@ func GetIPFamilyFromCidr(cidr string) string {
 
 	return addrType
 }
+
+func ipFamilyFromCIDRStrings(cidrs []string) (string, error) {
+	if len(cidrs) > 2 {
+		return InvalidIPFamily, errors.New("too many CIDRs specified")
+	}
+	var foundIPv4 bool
+	var foundIPv6 bool
+	for _, cidr := range cidrs {
+		cidrType := GetIPFamilyFromCidr(cidr)
+		if cidrType == IPv4IpFamily {
+			foundIPv4 = true
+		} else {
+			foundIPv6 = true
+		}
+	}
+	switch {
+	case foundIPv4 && foundIPv6:
+		return DualStackIPFamily, nil
+	case foundIPv4:
+		return IPv4IpFamily, nil
+	case foundIPv6:
+		return IPv6IpFamily, nil
+	default:
+		return InvalidIPFamily, nil
+	}
+}
+
+// GetClusterIPFamily returns a cluster IPFamily from the configuration provided.
+func GetClusterIPFamily(c *capi.Cluster) (string, error) {
+	var podCIDRs, serviceCIDRs []string
```

Comment:
Can we change following several lines into:

```go
podCIDRs := c.Spec.ClusterNetwork.Pods.CIDRBlocks
serviceCIDRs := c.Spec.ClusterNetwork.Services.CIDRBlocks
```
?
Will there be any issues?

Reply:
It is possible, but have you tried if that doesn't exist, what you will get if you call:
```go
podCIDRs := c.Spec.ClusterNetwork.Pods.CIDRBlocks
```

If it is empty array, there should be no issue?

Reply:
if we use like this, 
```
podCIDRs := c.Spec.ClusterNetwork.Pods.CIDRBlocks
serviceCIDRs := c.Spec.ClusterNetwork.Services.CIDRBlocks
```
unit tests will fail because of this error:
```
runtime error: invalid memory address or nil pointer dereference
```